src/main.rs

```diff
@@ -15,56 +15,90 @@ fn modload_handler(dbg: &mut Debugger, modname: &str, base: usize) {
         print!("    --freq               - \
                Treats all breakpoints as frequency breakpoints\n");
         print!("    --verbose            - \
                Enables verbose prints for debugging\n");
         print!("    --print              - \
                Prints breakpoint info on every single breakpoint\n");
+        print!("    --follow-fork        - \
+               Capture coverage for child processes\n");
         print!("    [explicit meso file] - \
                Load a specific meso file regardless of loaded modules\n\n");
 
-        print!("Standard usage: mesos.exe <pid>\n");
         return;
     }
 
-    // Attach to process
-    let mut dbg = Debugger::attach(args[1].parse().unwrap());
 
-    // Register callback routine for module loads so we can attempt to apply
-    // breakpoints to it from the meso file cache
-    dbg.register_modload_callback(Box::new(modload_handler));
+    let mut pid: i32 = -1;
+    let mut frequency_mode_enabled = false;
+    let mut verbose_mode_enabled = false;
+    let mut follow_fork_enabled = false;
+    let mut print_breakpoints_enabled = false;
+    let mut mesos: Vec<&Path> = Vec::new();
+
+    let mut argv: Vec<String> = Vec::new();
 
-    // Process arguments
     if args.len() > 2 {
-        for meso in &args[2..] {
-            if meso == "--freq" {
-                // Always do frequency
-                print!("Always frequency mode enabled\n");
-                dbg.set_always_freq(true);
-                continue;
-            } else if meso == "--verbose" {
-                // Enable verbose mode
-                print!("Verbose mode enabled\n");
-                dbg.set_verbose(true);
-                continue;
-            } else if meso == "--print" {
-                // Enable breakpoint print mode
-                print!("Breakpoint print mode enabled\n");
-                dbg.set_bp_print(true);
-                continue;
+        for (i, arg) in args[1..].iter().enumerate() {
+            if arg == "-p" {
+                pid = arg.parse().unwrap();
+            }
+            else if arg == "--verbose" {
+                verbose_mode_enabled = true;
+            }
+            else if arg == "--print" { 
+                print_breakpoints_enabled = true;
+            }
+            else if arg == "--freq" { 
+                frequency_mode_enabled = true;
+            }
+            else if arg == "--follow-fork" {
+                follow_fork_enabled = true;
+            }
+            else if arg == "--" {
+                argv.extend_from_slice(&args[i + 2..]);
+                break;
+            }
+            else { // Has to be a mesofile
+                //mesofile::load_meso(&mut dbg, Path::new(arg));
+                mesos.push(Path::new(arg));
             }
-
-            // Load explicit meso file
-            mesofile::load_meso(&mut dbg, Path::new(meso));
         }
     }
 
+    let mut dbg:Debugger;
+    if pid == -1 && argv.len() > 0 {
```

Comment:
Once `pid` is an `Option` this will be an `if pid.is_none()` rather than `pid == -1`